doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,484 @@
+     * @flags CREATE_AND_SET
+     * @default SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_E or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_X or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_T
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_FLAVOR,
+
+    /**
+     * @brief Packet action
+     *
+     * @type sai_packet_action_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PACKET_ACTION_FORWARD
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_PACKET_ACTION,
+
+    /**
+     * @brief Packet priority for trap/log actions
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_TRAP_PRIORITY,
+
+    /**
+     * @brief Next hop for cross-connect function
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_NEXT_HOP, SAI_OBJECT_TYPE_NEXT_HOP_GROUP, SAI_OBJECT_TYPE_ROUTER_INTERFACE
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_X
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_NEXT_HOP_ID,
+
+    /**
+     * @brief VRF for decapsulation and specific table lookup functions
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_T or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT4 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT6 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT46
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_VRF,
+
+    /**
+     * @brief Attach a counter
+     *
+     * When it is empty, then packet hits won't be counted
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_COUNTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_COUNTER_ID,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_END,
+
+    /** Custom range base value */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_END
+
+} sai_local_sid_entry_attr_t;
+
+/**
+ * @brief Local SID Entry
+ */
+typedef struct _sai_local_sid_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Virtual Router ID
+     *
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     */
+    sai_object_id_t vr_id;
+
+    /**
+     * @brief Length of the Locator part of the SID
+     */
+    sai_uint8_t locator_len;
```

Comment:
Can we define anything about SID stcuture in a separate object/container (sid_struct_t) -> locator_len, func_leb, args_len.

This way, this common struct could be re-used at places as/when required.

Reply:
@skraza Do you have any references or examples (VPP, Linux, FRR etc) of specifying the SID structure for anything other than the endpoint functionality? I'm trying to understand the implications of it on the hardware and also on proposed SONiC implementation of SRv6.

Reply:
If required, will be implemented in a subsequent PR.